src/audio/module_adapter/module_adapter.c

```diff
@@ -799,7 +799,7 @@ static int module_adapter_simple_copy(struct comp_dev *dev)
 
 		if (!mod->skip_sink_buffer_writeback)
 			buffer_stream_writeback(sink_c, mod->output_buffers[i].size);
-		comp_update_buffer_produce(sink_c, mod->output_buffers[i].size);
+		audio_stream_produce(&sink_c->stream, mod->output_buffers[i].size);
```

Comment:
I checked comp_update_buffer_consume and comp_update_buffer_produce, seems callback can be removed.
However, QB have failure, not sure it is caused by this part of change or not, maybe add:
if (mod->input_buffers[i].consumed)

Reply:
@btian1 can you take ownership of this PR and make sure it can pass testing. Thanks